src/arith/iter_affine_map.cc

```diff
@@ -2061,6 +2061,9 @@ class IterMapToExprNormalizer : public ExprMutator {
     if (analyzer_->CanProve(expr->extent == expr->source->extent) && is_one(expr->lower_factor)) {
       return source * expr->scale;
     } else if (analyzer_->CanProve(expr->source->extent == expr->lower_factor * expr->extent)) {
+      if (is_one(expr->extent) && !is_one(expr->lower_factor)) {
```

Comment:
would be nice ot have a rationale here

Reply:
I think it is better to use

```c++
`is_one(expr->extent) && !is_one(expr->source->extent)`
```

This will cover all the non-trivial iterator case